src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,550 @@
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character's Chocobo is able to Bore					*
+*																		*
+************************************************************************/
+
+	bool CanPlayerBore(CCharEntity* PChar)
+	{
+		// Check for Chocobo Bore ability here
+		return false;
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character has the Egg Helm equipped					*
+*																		*
+************************************************************************/
+
+	bool EggHelm(CCharEntity* PChar)
+	{
+		CItemArmor* PHead = (CItemArmor*)PChar->getStorage(LOC_INVENTORY)->GetItem(PChar->equip[SLOT_HEAD]);
+		if(PHead->getID() == 16109)
+		{
+			return true;
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+/************************************************************************
+*																		*
+*	Check if the character is digging at Night Time						*
+*																		*
+************************************************************************/
+
+	bool NightDigging()
+	{
+		TIMETYPE VanadielTOTD = CVanaTime::getInstance()->SyncTime();
+		if (VanadielTOTD == TIME_NIGHT)
+		{
+			return true;
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+/************************************************************************
+*																		*
+*	Get / Set the time of the last Chocobo Digging attempt				*
+*																		*
+************************************************************************/
+
+	int32 SetLastDig(CCharEntity* PChar)
+	{
+		uint32 CurrentTime = CVanaTime::getInstance()->getVanaTime();
+		Sql_Query(SqlHandle, "REPLACE INTO char_vars (charid, varname, value) VALUES ('%u', 'Last_Dig_Time', '%u');", PChar->id, CurrentTime); 
+		return 1;
+	}
+
+	int32 GetLastDig(CCharEntity* PChar)
+	{
+		const int8* fmtQuery = "SELECT value FROM char_vars WHERE charid = %u AND varname = 'Last_Dig_Time' LIMIT 1;";
+		int32 ret = Sql_Query(SqlHandle,fmtQuery,PChar->id);
+		int32 value = 0;
+
+		if (ret != SQL_ERROR &&
+			Sql_NumRows(SqlHandle) != 0 &&
+			Sql_NextRow(SqlHandle) == SQL_SUCCESS)
+		{
+			value = (int32)Sql_GetIntData(SqlHandle,0);
+		}
+		return value;
+	}
+
+/************************************************************************
+*																		*
+*	Update the Chocobo Digging delay									*
+*																		*
+************************************************************************/
+
+	void UpdateDigDelay(CCharEntity* PChar)
+	{
+	}
+
+/************************************************************************
+*																		*
+*	Return the Chocobo Digging delay									*
+*																		*
+************************************************************************/
+
+	int8 ReturnDigDelay(CCharEntity* PChar)
+	{
+		uint8 delay		= 16;								// Base Delay of 16
+		uint8 skillRank = PChar->RealSkills.rank[58];		// Grab Digging Rank
+		switch(skillRank)
+		{
```

Comment:
should be able to just         delay = dsp_cap(delay - (skillRank * 5), 0, 16); or something instead of using a switch here